Use Tesla authentication method from settings

hobbe opened the pull request:
Add email and password as app settings.
Use email/password to get a token, store the token as app setting.

If existing token is null or rejected, create a new token.

To create a token, use Tesla's email/password authentication with email/password from app settings.
But if email or password is not set in app settings, use website authentication on phone.
In any case, the token created is saved as an app setting for the next sessions.

![image](https://user-images.githubusercontent.com/312886/71323438-6d611600-24d3-11ea-9257-4bcf21f7421b.png)

hobbe committed:
Use Tesla authentication method from settings

Add email and password as app settings
Use email/password to get a token, store the token as app setting
If existing token is null or rejected, create a new token

hobbe committed:
Refactor authenticate method

srwalter reviewed (commented):
I don't want to replace the existing OAuth flow with an email/password based one, but I would accept having it as an option.  For example, if the email and password settings are set, then use them, but otherwise fall back to the current OAuth flow.

As I've stated elsewhere, I view the settings-based authentication flow as being much more cumbersome, and only marginally more trustworthy.  You still have to trust the app to guard the credentials, and now the username and password is being stored as a setting, rather than just a token that expires.

hobbe committed:
Refactor settings to return null is property is not set

hobbe committed:
Re-add website authentication as fallback

If token is not set, use Tesla's email/password authentication
If email or password is not set, use website authentication on phone

hobbe commented:
Added the original website authentification from phone back, as a fallback if user credentials are not provided as app settings.
